src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -169,6 +170,11 @@ public JsonRpc(Stream sendingStream, Stream receivingStream, object target = nul
         public JsonRpc(IJsonRpcMessageHandler messageHandler, object target)
             : this(messageHandler)
         {
+            if (messageHandler.Formatter is IJsonRpcInstanceContainer formatter)
+            {
+                formatter.Rpc = this;
+            }
+
             if (target != null)
```

Comment:
This should go into the next ctor overload down so that it _always_ runs rather than only when a local target object is given to the ctor.